starcluster/balancers/sge/__init__.py

```diff
@@ -36,6 +37,11 @@
 DEFAULT_STATS_FILE = os.path.join(DEFAULT_STATS_DIR, 'sge-stats.csv')
 
 
+class SGEJob(dict):
+    def __hash__(self):
+        return int(self['JB_job_number'])
```

Comment:
It would be a bug in OGS or in the qstat command. There should always be a JB_job_number.